test/functional/autocmd/termclose_spec.lua

```diff
@@ -14,13 +14,38 @@ describe('TermClose event', function()
     nvim('set_option', 'shellcmdflag', 'EXE')
   end)
 
-  it('triggers when terminal job ends', function()
+  it('triggers when fast-exiting terminal job stops', function()
     command('autocmd TermClose * let g:test_termclose = 23')
     command('terminal')
     command('call jobstop(b:terminal_job_id)')
     retry(nil, nil, function() eq(23, eval('g:test_termclose')) end)
   end)
 
+  it('triggers when long-running terminal job gets stopped', function()
+    nvim('set_option', 'shell', 'sh')
+    command('autocmd TermClose * let g:test_termclose = 23')
+    command('terminal')
+    command('call jobstop(b:terminal_job_id)')
+    retry(nil, nil, function() eq(23, eval('g:test_termclose')) end)
+  end)
+
+  it('triggers when long-running terminal job gets stopped (trapping SIGHUP)', function()
+    nvim('set_option', 'shell', 'sh')
+    nvim('set_option', 'shellcmdflag', '-c')
+    command('autocmd TermClose * let g:test_termclose = 23')
+    command('autocmd TermResponse * let g:test_termopen = 1')
+
+    command('terminal trap "" HUP && echo SETUP TRAP >> ~/.nvimlog && sleep 5')
+    command('sleep 200m')
+    -- command('terminal')
+    -- command('call jobsend(b:terminal_job_id, ["trap \'\' HUP", "(echo $$; trap;) >> ~/.nvimlog", "sleep 5"])')
+    -- eq(2, eval('b:terminal_job_pid'))
```

Comment:
Any comments about this?
I would like to avoid the sleep here, but ensure that the terminal is up and running (having setup the trap).
/cc @justinmk